### PR TITLE
Some fixes in Python fuzzer logic

### PR DESCRIPTION
**Pull request by tochilinak**
## Description

Changes:
- New logic for functions with default arguments: first try to fuzz only positional arguments, then all of them.
- Added new Python tree node: FakeNode (instead of Java `null`). If fuzzer provides object with this node, we don't start execution with this data.
- BaselineAlgorithm now provides random type each 6 iterations.
- Logic for abstract classes is now the same as for protocols.

### Manual tests

Default arguments: https://github.com/TheAlgorithms/Python/blob/master/machine_learning/gradient_descent.py (function `_error`).

Random type: https://github.com/TheAlgorithms/Python/blob/master/data_structures/queue/circular_queue.py  (function `enqueue` of class `CircularQueue`). Mypy does not pass any types for `data` parameter, bacause it consideres `self.array` to be `list[None]`.

Test for abstract class:

```python
from typing import AbstractSet

def id_(x: AbstractSet[int]):
    return x
```

Generated output:
```python
import sys
sys.path.append(r'../../../utbot-python/samples/easy_samples')
import builtins
import general
import unittest


class TestTopLevelFunctions(unittest.TestCase):
    # region Test suites for executable general.id_
    
    # region FUZZER
    
    def test_id_(self):
        """
        x = builtins.set[typing.Any]
        """
        actual = general.id_(set())
        
        self.assertEqual(set(), actual)
    # endregion
    
    # endregion
```

## Self-check list

Check off the item if the statement is true. Hint: [x] is a marked item.

Please do not delete the list or its items.

- [x] I've set the proper **labels** for my PR (at least, for category and component).
- [x] PR **title** and **description** are clear and intelligible.
- [x] I've added enough **comments** to my code, particularly in hard-to-understand areas.
- [ ] The functionality I've repaired, changed or added is covered with **automated tests**.
- [x] **Manual tests** have been provided optionally.
- [ ] The **documentation** for the functionality I've been working on is up-to-date.

### CODE REVIEW

**Review by Markoutte — APPROVED** [review]

File: utbot-python/src/main/kotlin/org/utbot/python/PythonTestCaseGenerator.kt [hunk 1]
```diff
@@ -196,6 +196,40 @@ class PythonTestCaseGenerator(
...
+            val firstUntil = (until - now) / 2 + now
+            val originalDef = method.definition
+            val shortType = meta.removeNonPositionalArgs(originalDef.type)
+            val shortMeta = PythonFuncItemDescription(
+                originalDef.meta.name,
+                originalDef.meta.args.take(shortType.arguments.size)
+            )
+            val additionalVars = originalDef.meta.args.drop(shortType.arguments.size).fold("") { acc, arg ->
+                acc + "\n" + "${arg.name}: ${pythonAnyType.pythonTypeRepresentation()}"  // TODO: better types
+            }
```
Comment (tamarinvs19) [inline utbot-python/src/main/kotlin/org/utbot/python/PythonTestCaseGenerator.kt:225]: I suggest to simplify this code:
```kotlin
val additionalVars = originalDef.meta.args
                .drop(shortType.arguments.size)
                .joinToString(separator="\n", prefix="\n") { arg ->
                "${arg.name}: ${pythonAnyType.pythonTypeRepresentation()}"  // TODO: better types
            }
```

File: utbot-python/src/main/kotlin/org/utbot/python/PythonTestCaseGenerator.kt [hunk 2]
```diff
@@ -257,7 +292,9 @@ class PythonTestCaseGenerator(
                 getOffsetLine(sourceFileContent, method.ast.beginOffset),
                 getOffsetLine(sourceFileContent, method.ast.endOffset)
             ),
-            mypyConfigFile
+            mypyConfigFile,
+            additionalVars,
+            randomTypeFrequency = 6
```
Comment (tamarinvs19) [inline utbot-python/src/main/kotlin/org/utbot/python/PythonTestCaseGenerator.kt:297]: Maybe move 6 to constant?

Comment (tochilinak) [reply utbot-python/src/main/kotlin/org/utbot/python/PythonTestCaseGenerator.kt]: Fixed

File: utbot-python/src/main/kotlin/org/utbot/python/PythonTestCaseGenerator.kt (hunk 1 above)
Comment (tochilinak) [reply utbot-python/src/main/kotlin/org/utbot/python/PythonTestCaseGenerator.kt]: Fixed

**Review by tamarinvs19 — APPROVED** [review]

**Pull request merged**